scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -44,149 +49,233 @@ set.seed(12345)
-If this is a useful marker, we should see a bimodal plot of the marker density.
+To do this, we will need to make some decisions, like what thresholds should we use to determine whether a cell is or is not expressing a particular marker. 
+In general, if a marker is going to be useful for this type of assignment, we would like to see a bimodal plot for the marker in question that can be used to separate the population into two groups of cells based on the level of that marker.
 
-We'll do a bit of wrangling to pull out the ADT data, transpose it so each row is a cell,
-then convert it into a data frame for plotting. 
-(We have to first convert to a "regular" R matrix or R won't know how to convert it to a data frame)
+To look at the expression levels of each of the marker genes we are interested in, we will pull out the expression values for the markers from the SCE object and do some data wrangling. 
 
-```{r}
-# convert to a data frame
+The data we are interested is the normalized count data for the ADT tags, which is stored in the `logcounts` assay.
+If you recall, this matrix is stored with the columns as cells and rows as markers, but we really want it the other way, so we will first transpose the data, then convert it to a data frame for our next steps.
+Because the SCE object stores the main data matrices in a specialized format, we have to do one extra step convert it first to a "regular" R matrix or R won't know how to convert it to a data frame.
+
+```{r extract ADT}
+# convert logcounts data to a data frame
 adt_df <- logcounts(altExp(sce)) %>%
-  t() %>% 
-  as.matrix() %>%
-  as.data.frame()
+  t() %>% # transpose
+  as.matrix() %>% # convert to matrix
+  as.data.frame() # convert to data frame
+
+# view the data frame
+head(adt_df)
 ```
 
-With a bit more wrangling, we can put the expression data for the three tags we care about into a single column, which will allow us to plot all three together a bit more neatly.
-Here we are using the `tidyr::pivot_longer()` function, which allows us to convert our data frame with one column per tag into a data frame with separate columns for the tag id (`ADT`) and the expression value (`logcount`).
+If we just wanted to plot one of these tags, we could do so right away, but with a bit more data wrangling, we can convert these results into a "tidier" format, that will allow us to take full advantage of `tidyverse` tools!
 
-```{r}
+Right now the data is in a "wide" format, with each column a different tag. 
+But the data in all of the columns is the same type, and measures something similar: the normalized count of an ADT.
+One could even argue that each row contains 10 different observations, where the "tidy" data ideal, as espoused by [Wickham (2014)](https://doi.org/10.18637/jss.v059.i10), requires a single observation per row, a "long" format.
+This long format will have one column that tells us which ADT was measured and a second column with the measurement value itself.
```

Comment:
Something worth pointing with reshaping (might even be worth writing out explicitly somehow) is that folks newer to tidy formats are often sad at the "repetition" in the resulting long df, e.g. multiple rows for the same cell. We'll want to point out that this is FINE and NOT REDUNDANT and actually can make some tasks EASIER TO DO. 

In this case we don't actually have a cell_id column though so they might not have this explicit confusion in this case.